src/window/createStatWidget.ts

```diff
@@ -0,0 +1,81 @@
+import * as events from "../helpers/events";
+import { getElementVisibility } from "../helpers/flexHelper";
+import {
+  WidgetCreator,
+  FlexiblePosition,
+  groupbox,
+  horizontal,
+  label,
+  compute,
+  Store,
+} from "openrct2-flexui";
+
+/**
+ * Creates a stat widget with the given properties.
+ *
+ * @param {string} props.title - The title of the stat widget.
+ * @param {Store<T>} props.gameStatStore - The statistic store with values for the entire game.
+ * @param {Store<T>} props.parkStatStore - The park stat store with values for the current park.
+ * @param {(stat: T) => string} [props.processStat] - Optional function to process the stat value for displaying it more clearly.
+ * @returns {WidgetCreator<FlexiblePosition>} The created stat widget.
+ */
+export function createStatWidget<T>(props: {
+  /** The title of your widget which will display in the box*/
+  title: string;
+  /** The store with total game stat values*/
+  gameStatStore: Store<T>;
+  /** The store with this park stat values*/
+  parkStatStore: Store<T>;
+  /** Opional function to process the stat value,
+   * e.g. to process time in seconds into nice human-readable text,
+   * or to take the take the length of an array instead of its values*/
+  processStat?: (stat: T) => string;
```

Comment:
I went back and forth on this. Making it optional like a possible way to decrease the initial lift of creating a new stat. hence why I added the JSON.stringify fallback. I think this is easy enough that instead we can add a few examples of what this could look like (which are actually already done pretty well with the time example and the two that i created) and trust the user to get it.